dev/com.ibm.ws.jaxws.2.3.common/src/com/ibm/ws/jaxws/client/LibertyServiceImpl.java

```diff
@@ -158,6 +159,8 @@ protected void configureCustomizeBinding(Client client, QName portName) {
 
         Set<ConfigProperties> configPropsSet = servicePropertiesMap.get(portName);
         client.getOutInterceptors().add(new LibertyCustomizeBindingOutInterceptor(wsrInfo, securityConfigService, configPropsSet));
+
+        client.getOutInterceptors().add(new LibertyJaxWsClientSSLOutInterceptor(wsrInfo, securityConfigService, configPropsSet, client.getEndpoint().getEndpointInfo()));
```

Comment:
I hadn't considered this. I borrowed the SSLOutInterceptor model from the jaxrs-2.1 implementation, but I recycled the original SSL config code that we had in the LibertyCustomizeBindingOutInterceptor. That class has a conditional check that I moved over into the LibertyJaxWsClientSSLOutInterceptor, it's handleMessage() method won't do anything if there's no SSL Config, just like was being done in LibertyCustomizeBindingOutInterceptor. There might be a case for moving the check to registering the interceptor, rather than always enabling it...

Reply:
I forgot to put a link to the JAXRS way of doing it:  https://github.com/OpenLiberty/open-liberty/blob/integration/dev/com.ibm.ws.jaxrs.2.0.client/src/com/ibm/ws/jaxrs20/client/JAXRSClientImpl.java#L193

Reply:
So JAX-RS is using the Config to check whether on not to add the SSL Interceptor, which works on one level but the LibertyJaxWsClientSecurityOutInterceptor (I've renamed it like you suggested) does more than just SSL, like you rightly pointed out, so I do think we should always enable it. I could refactor the interceptor into multiple interceptors for handling SSL, Basic Auth, and WSDL redirects but I'm not convinced that's necessary.